exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -505,24 +507,26 @@ interface ISqlExpressionBuilder {
         Between(this, wrap(EntityID(from, this.idTable())), wrap(EntityID(to, this.idTable())))
 
     /** Returns `true` if this expression is null, `false` otherwise. */
-    fun <T> Expression<T>.isNull() = if (this is Column<*> && isEntityIdentifier()) {
-        table.mapIdOperator(::IsNullOp)
-    } else {
-        IsNullOp(this)
+    fun <T> Expression<T>.isNull() = when {
+        this is Column<*> && isEntityIdentifier() -> table.mapIdOperator(::IsNullOp)
+        this is QueryParameter && compositeValue != null -> {
+            val table = compositeValue.values.keys.first().table
+            table.mapIdOperator(::IsNullOp)
+        }
+        else -> IsNullOp(this)
     }
 
     /** Returns `true` if this string expression is null or empty, `false` otherwise. */
-    fun <T : String?> Expression<T>.isNullOrEmpty() = if (this is Column<*> && isEntityIdentifier()) {
-        table.mapIdOperator(::IsNullOp)
-    } else {
-        IsNullOp(this)
-    }.or { this@isNullOrEmpty.charLength() eq 0 }
+    fun <T : String?> Expression<T>.isNullOrEmpty() = IsNullOp(this).or { this@isNullOrEmpty.charLength() eq 0 }
```

Comment:
Not sure why this logic was added, as the function signature means it's not possible to call `isNullOrEmpty()` on an `IdTable.id`, so the first branch will always be false. `charLength()` also cannot be invoked on `IdTable.id`.